### ISSUE TITLE
Pre-trained weights on the KITTI dataset

### ISSUE THREAD

**Issue by AlanLiangC** [Say2L/DCDet#1]
Hello, thank you for your valuable contribution. Could you please provide pre-trained weights on the KITTI dataset? I obtained results far below the reported 85.82 in the paper, and I'm not sure what the issue might be. My results are as follows.

2024-02-29 10:48:32,845   INFO  Generate label finished(sec_per_example: 0.0136 second).
2024-02-29 10:48:32,845   INFO  recall_roi_0.3: 0.000000
2024-02-29 10:48:32,845   INFO  recall_rcnn_0.3: 0.908131
2024-02-29 10:48:32,846   INFO  recall_roi_0.5: 0.000000
2024-02-29 10:48:32,846   INFO  recall_rcnn_0.5: 0.899931
2024-02-29 10:48:32,846   INFO  recall_roi_0.7: 0.000000
2024-02-29 10:48:32,846   INFO  recall_rcnn_0.7: 0.772967
2024-02-29 10:48:32,847   INFO  Average predicted number of objects(3769 samples): 4.315
2024-02-29 10:48:52,948   INFO  Car AP@0.70, 0.70, 0.70:
bbox AP:90.8334, 90.0609, 89.5380
bev  AP:90.2614, 88.5962, 87.7556
3d   AP:89.3006, 79.5609, 78.7285
aos  AP:90.82, 90.02, 89.46
Car AP_R40@0.70, 0.70, 0.70:
bbox AP:96.7571, 95.5560, 93.0914
bev  AP:93.6016, 91.7121, 89.3786
3d   AP:92.2752, 83.7243, 81.0321
aos  AP:96.74, 95.49, 92.99
Car AP@0.70, 0.50, 0.50:
bbox AP:90.8334, 90.0609, 89.5380
bev  AP:90.8229, 90.1188, 89.6828
3d   AP:90.8229, 90.0976, 89.6413
aos  AP:90.82, 90.02, 89.46
Car AP_R40@0.70, 0.50, 0.50:
bbox AP:96.7571, 95.5560, 93.0914
bev  AP:96.7801, 95.8243, 95.3667
3d   AP:96.7699, 95.7734, 93.2428
aos  AP:96.74, 95.49, 92.99

**Comment by Say2L** [comment]
> Hello, thank you for your valuable contribution. Could you please provide pre-trained weights on the KITTI dataset? I obtained results far below the reported 85.82 in the paper, and I'm not sure what the issue might be. My results are as follows.
> 
> 2024-02-29 10:48:32,845 INFO Generate label finished(sec_per_example: 0.0136 second). 2024-02-29 10:48:32,845 INFO recall_roi_0.3: 0.000000 2024-02-29 10:48:32,845 INFO recall_rcnn_0.3: 0.908131 2024-02-29 10:48:32,846 INFO recall_roi_0.5: 0.000000 2024-02-29 10:48:32,846 INFO recall_rcnn_0.5: 0.899931 2024-02-29 10:48:32,846 INFO recall_roi_0.7: 0.000000 2024-02-29 10:48:32,846 INFO recall_rcnn_0.7: 0.772967 2024-02-29 10:48:32,847 INFO Average predicted number of objects(3769 samples): 4.315 2024-02-29 10:48:52,948 INFO Car AP@0.70, 0.70, 0.70: bbox AP:90.8334, 90.0609, 89.5380 bev AP:90.2614, 88.5962, 87.7556 3d AP:89.3006, 79.5609, 78.7285 aos AP:90.82, 90.02, 89.46 Car AP_R40@0.70, 0.70, 0.70: bbox AP:96.7571, 95.5560, 93.0914 bev AP:93.6016, 91.7121, 89.3786 3d AP:92.2752, 83.7243, 81.0321 aos AP:96.74, 95.49, 92.99 Car AP@0.70, 0.50, 0.50: bbox AP:90.8334, 90.0609, 89.5380 bev AP:90.8229, 90.1188, 89.6828 3d AP:90.8229, 90.0976, 89.6413 aos AP:90.82, 90.02, 89.46 Car AP_R40@0.70, 0.50, 0.50: bbox AP:96.7571, 95.5560, 93.0914 bev AP:96.7801, 95.8243, 95.3667 3d AP:96.7699, 95.7734, 93.2428 aos AP:96.74, 95.49, 92.99

Thanks for your attention. You can download the KITTI pretrained model from here https://drive.google.com/file/d/1hD7cgme5y-AHP7V7A24UTwMo_tAagCG2/view?usp=drive_link. The KITTI dataset is too small and may lead to unstable results. You can try the pretrained model and keep the default config parameters unchanged.

**Comment by Say2L** [comment]
> Hello, thank you for your valuable contribution. Could you please provide pre-trained weights on the KITTI dataset? I obtained results far below the reported 85.82 in the paper, and I'm not sure what the issue might be. My results are as follows.
> 
> 2024-02-29 10:48:32,845 INFO Generate label finished(sec_per_example: 0.0136 second). 2024-02-29 10:48:32,845 INFO recall_roi_0.3: 0.000000 2024-02-29 10:48:32,845 INFO recall_rcnn_0.3: 0.908131 2024-02-29 10:48:32,846 INFO recall_roi_0.5: 0.000000 2024-02-29 10:48:32,846 INFO recall_rcnn_0.5: 0.899931 2024-02-29 10:48:32,846 INFO recall_roi_0.7: 0.000000 2024-02-29 10:48:32,846 INFO recall_rcnn_0.7: 0.772967 2024-02-29 10:48:32,847 INFO Average predicted number of objects(3769 samples): 4.315 2024-02-29 10:48:52,948 INFO Car AP@0.70, 0.70, 0.70: bbox AP:90.8334, 90.0609, 89.5380 bev AP:90.2614, 88.5962, 87.7556 3d AP:89.3006, 79.5609, 78.7285 aos AP:90.82, 90.02, 89.46 Car AP_R40@0.70, 0.70, 0.70: bbox AP:96.7571, 95.5560, 93.0914 bev AP:93.6016, 91.7121, 89.3786 3d AP:92.2752, 83.7243, 81.0321 aos AP:96.74, 95.49, 92.99 Car AP@0.70, 0.50, 0.50: bbox AP:90.8334, 90.0609, 89.5380 bev AP:90.8229, 90.1188, 89.6828 3d AP:90.8229, 90.0976, 89.6413 aos AP:90.82, 90.02, 89.46 Car AP_R40@0.70, 0.50, 0.50: bbox AP:96.7571, 95.5560, 93.0914 bev AP:96.7801, 95.8243, 95.3667 3d AP:96.7699, 95.7734, 93.2428 aos AP:96.74, 95.49, 92.99

You can download the latest dynamic_cross_head_v1.py file. The KITTI result reported in the paper was obtained by this version presented in the repo now. We use an extra post-processing method proposed by our another repo NIV-SSD.
